expression/builtin_time.go

```diff
@@ -2657,7 +2672,18 @@ func (du *baseDateArithmitical) add(ctx sessionctx.Context, date types.Time, int
 
 	duration := time.Duration(dur)
 	goTime = goTime.Add(duration)
-	goTime = goTime.AddDate(int(year), int(month), int(day))
+	// when we execute select date_add('2018-01-31',interval 1 month) in mysql we got 2018-02-28
+	// but in tidb we got 2018-03-03
+	// dig it and we found it's caused by golang api time.DateDate(year int, month Month, day, hour, min, sec, nsec int, loc *Location) Time ,
```

Comment:
Please uppercase the first letter of each comment sentence.

Reply:
@imtbkcat PTAL